sdks/go/pkg/beam/io/xlang/kafkaio/kafka.go

```diff
@@ -0,0 +1,288 @@
+//
+// Note that the "bootstrap.servers" property is automatically set by
+// kafkaio.Read and does not need to be specified via this option.
+func ConsumerConfigs(cfgs map[string]string) readOption {
+	return func(cfg *readConfig) {
+		for k, v := range cfgs {
+			cfg.pl.ConsumerConfig[k] = v
+		}
+	}
+}
+
+// StartReadTimestamp is a Read option that specifies a start timestamp in
+// milliseconds epoch, so only records after that timestamp will be read.
+//
+// This results in failures if one or more partitions don't contain messages
+// with a timestamp larger than or equal to the one specified, or if the
+// message format version in a partition is before 0.10.0, meaning messages do
+// not have timestamps.
+func StartReadTimestamp(ts int64) readOption {
+	return func(cfg *readConfig) {
+		cfg.pl.StartReadTime = &ts
+	}
+}
+
+// MaxNumRecords is a Read option that specifies the maximum amount of records
+// to be read. Setting this will cause the Read to execute as a bounded
+// transform. Useful for tests tests and demo applications.
+func MaxNumRecords(num int64) readOption {
+	return func(cfg *readConfig) {
+		cfg.pl.MaxNumRecords = &num
+	}
+}
+
+// MaxReadSecs is a Read option that specifies the maximum amount of time in
+// seconds the transform executes. Setting this will cause the Read to execute
+// as a bounded transform. Useful for tests and demo applications.
+func MaxReadSecs(secs int64) readOption {
+	return func(cfg *readConfig) {
+		cfg.pl.MaxReadTime = &secs
+	}
+}
+
+// CommitOffsetInFinalize is a Read option that specifies whether to commit
+// offsets when finalizing.
+//
+// Default: false
+func CommitOffsetInFinalize(enabled bool) readOption {
+	return func(cfg *readConfig) {
+		cfg.pl.CommitOffsetInFinalize = enabled
+	}
+}
+
+// TimestampPolicy is a Read option that specifies the timestamp policy to use
+// for extracting timestamps from the KafkaRecord. Must be one of the predefined
+// constant timestamp policies in this package.
+//
+// Default: kafkaio.ProcessingTime
+func TimestampPolicy(name policy) readOption {
+	return func(cfg *readConfig) {
+		cfg.pl.TimestampPolicy = string(name)
+	}
+}
+
+// readPayload should produce a schema matching the expected cross-language
+// payload for Kafka reads. An example of this on the receiving end can be
+// found in the Java SDK class
+// org.apache.beam.sdk.io.kafka.KafkaIO.Read.External.Configuration.
+type readPayload struct {
+	ConsumerConfig         map[string]string
+	Topics                 []string
+	KeyDeserializer        string
+	ValueDeserializer      string
+	StartReadTime          *int64
+	MaxNumRecords          *int64
+	MaxReadTime            *int64
+	CommitOffsetInFinalize bool
+	TimestampPolicy        string
+}
+
+// Write is a cross-language PTransform which writes KV data to a specified
+// Kafka topic. By default, this assumes keys and values to be received as raw
+// byte arrays. This can be changed through optional parameters.
```

Comment:
```suggestion
// Kafka topic. By default, this assumes keys and values to be received as 
// byte slices. This can be changed through optional parameters.
```